### PR TITLE
Remove links to the remote extension registry

### PR DESCRIPTION
**Pull request by taras-yemets**
Removes references to _sourcegraph.com/extensions_ as it's no longer available.

## Test plan
Existing checks in the CI pass.

### CODE REVIEW

File: doc/admin/extensions/index.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Administration of Sourcegraph extensions and the extension registry
 
-[Sourcegraph extensions](../../extensions/index.md) add features to Sourcegraph. Sourcegraph Free and Enterprise instances allow users to view and enable extensions from the [Sourcegraph.com extension registry](https://sourcegraph.com/extensions).
+[Sourcegraph extensions](../../extensions/index.md) add features to Sourcegraph. Sourcegraph Free and Enterprise instances allow users to view and enable extensions from the sourcegraph.com extension registry.
```
Comment (taras-yemets) [inline doc/admin/extensions/index.md:3]: Hereinafter we replace the link to the sourcegraph.com extension registry text. Does the text remain valuable to users with this change? What do you think, @ryankscott?

File: doc/admin/how-to/troubleshoot-sg-extension.md [hunk 2]
```diff
@@ -126,12 +126,12 @@ The ESlint extension requires the eslint.insight.repository and eslint.insight.s
...
 2. Once it is enabled, you will find an `Export to CSV` button in the Search-Results page
  
 #### Search-export: Network Error when downloading CSV
 
-It's likely that the CSV file exceeds the browser's limit for data URI size. Users can limit the size of search match preview size through their user settings (see ["contributions"](https://sourcegraph.com/extensions/sourcegraph/search-export/-/contributions) for search-export). If decreasing the size of search match previews doesn't resolve the issue, users can decrease the amount of search results exported with the `count:` filter in their search query.
+It's likely that the CSV file exceeds the browser's limit for data URI size. Users can limit the size of search match preview size through their user settings (see ["contributions"](https://github.com/sourcegraph/sourcegraph-search-export/blob/master/package.json#L26-L66) for search-export). If decreasing the size of search match previews doesn't resolve the issue, users can decrease the amount of search results exported with the `count:` filter in their search query.
```
Comment (taras-yemets) [inline doc/admin/how-to/troubleshoot-sg-extension.md:134]: We'll probably remove this search-export extension related questions after we migrate it to the core workflow.
For now, we replace links to the extension registry with the ones to the extension's repository.

File: doc/adopt/code_search_in_monorepos.md [hunk 3]
```diff
@@ -45,7 +45,7 @@ It might *seem* like staying in a single tool (your editor) means staying in flo
...
-- 😊 The code search tool can show much more helpful code context, such as [Git blame information after each line](https://sourcegraph.com/extensions/sourcegraph/git-extras), [code coverage overlays](https://sourcegraph.com/extensions/sourcegraph/codecov), runtime info from [Datadog](https://sourcegraph.com/extensions/sourcegraph/datadog-metrics)/[LightStep](https://sourcegraph.com/extensions/sourcegraph/lightstep)/[Sentry](https://sourcegraph.com/extensions/sourcegraph/sentry)/etc., static analysis and lint results from [SonarQube](https://sourcegraph.com/extensions/sourcegraph/sonarqube), and more. You could configure some of these things to display in your editor, but that's cumbersome and they're noisy for the majority of the time when you're writing code.
+- 😊 The code search tool can show much more helpful code context, such as git blame information after each line, code coverage overlays, runtime info from Datadog/LightStep/Sentry/etc., static analysis and lint results from SonarQube, and more. You could configure some of these things to display in your editor, but that's cumbersome and they're noisy for the majority of the time when you're writing code.
```
Comment (taras-yemets) [inline doc/adopt/code_search_in_monorepos.md:48]: Just removing links for now.
It looks like we're going to delete/edit this list item once we sunset extensions.

File: doc/code_intelligence/explanations/search_based_code_intelligence.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ Search-based code navigation also filters results by file extension and by impor
 
 ## What languages are supported?
 
-Search-based code navigation supports all of [the most popular programming languages](https://sourcegraph.com/extensions?category=Programming+languages).
+Search-based code navigation supports all of [the most popular programming languages](https://github.com/sourcegraph/code-intel-extensions/blob/master/template/src/language-specs/languages.ts#L353-L399).
```
Comment (taras-yemets) [inline doc/code_intelligence/explanations/search_based_code_intelligence.md:19]: Do we have a better place where all the supported languages are listed (see also [this comment](https://github.com/sourcegraph/sourcegraph/pull/39928/files#r937687712))?

File: doc/integration/index.md [hunk 5]
```diff
@@ -17,11 +17,6 @@ Sourcegraph integrates with your other tools to help you search, navigate, and r
   - [Python dependencies](python.md)
   - [Go dependencies](go.md)
   - [Other Git repository hosts](../admin/external_service/other.md)
-- Other services
-  - [Codecov](https://sourcegraph.com/extensions/sourcegraph/codecov)
-  - [Datadog](https://sourcegraph.com/extensions/sourcegraph/datadog-metrics)
-  - [LightStep](lightstep.md)
-  - [View all "External services" extensions](https://sourcegraph.com/extensions?query=category%3A%22External+services%22)
```
Comment (taras-yemets) [inline doc/integration/index.md:24]: Removing extensions mentions and dedicated Lightstep docs page.

**Comment by philipp-spiess** [comment]
Do we still want to go ahead with this? `sourcegraph.com/extensions` not being available as an oversight :/

**Pull request closed without merging**